src/Cygni.Snake.Client/Map.cs

```diff
@@ -63,28 +87,49 @@ public IEnumerable<MapCoordinate> SnakeParts
             }
         }
 
+        public DirectionalResult GetResultOfMyDirection(Direction dir)
+        {
+            var target = MySnake.HeadPosition.GetDestination(dir);
+
+            switch (this[target])
+            {
+                case TileType.Food:
+                    return DirectionalResult.Food;
+                case TileType.Empty:
+                case TileType.OpponentHeadNeighbor:
+                    return DirectionalResult.Nothing;
+                case TileType.OpponentTail:
+                    return Tick % 3 == 0 ? DirectionalResult.TailNibble : DirectionalResult.Death;
```

Comment:
I **think** tail growth is dependent on a game setting, so not sure `Tick % 3 == 0` works here as a general solution.